e2e-saas/utils/RhCheReporter.ts

```diff
@@ -0,0 +1,171 @@
+/*********************************************************************
+ * Copyright (c) 2019 Red Hat, Inc.
+ *
+ * This program and the accompanying materials are made
+ * available under the terms of the Eclipse Public License 2.0
+ * which is available at https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ **********************************************************************/
+import * as mocha from 'mocha';
+import { TYPES, IDriver, DriverHelper, CLASSES, ScreenCatcher, ITestWorkspaceUtil, PreferencesHandler, TestConstants, AskForConfirmationType, CheApiRequestHandler } from 'e2e';
+import { logging } from 'selenium-webdriver';
+import fs from 'fs';
+import rm from 'rimraf';
+import { rhcheContainer } from '../inversify.config';
+
+const driver: IDriver = rhcheContainer.get(TYPES.Driver);
+const driverHelper: DriverHelper = rhcheContainer.get(CLASSES.DriverHelper);
+const screenCatcher: ScreenCatcher = rhcheContainer.get(CLASSES.ScreenCatcher);
+let methodIndex: number = 0;
+let deleteScreencast: boolean = true;
+let testWorkspaceUtil: ITestWorkspaceUtil = rhcheContainer.get(TYPES.WorkspaceUtil);
+let preferencesHalder: PreferencesHandler = rhcheContainer.get(CLASSES.PreferencesHandler);
+
+class RhCheReporter extends mocha.reporters.Spec {
```

Comment:
@Katka92 @ScrewTSW can we avoid using this reporter in 7.11.0 and fall back on upstream one ? e.g. remove this hot-fix for tests in the upstream for preference updates? cc: @rhopp

Reply:
I'm not sure I can answer this. @Katka92 has better knowledge of the reporter classes.

Reply:
the goal to get rid of downstream UD patches, custom reporters etc. We should fully reuse upstream tests

Reply:
@ibuziuk The main problem which RhCheReporter is solving is that if we use upstream reporter, the inversion of control uses all upstream classes, even for obtaining user token. We need to inject the RhCheSpecific classes prior to anything else happens, and the first class that is executed during the test run is the reporter. So, this is not just a hot-fix for preferences handler but also fix for all API calls using a token on the rh-che side (like setting up a preferences).